deltametrics/mask.py

```diff
@@ -48,94 +46,823 @@ def show(self, **kwargs):
+        >>> velocity = rcm8cube['velocity'][-1, :, :]
+        >>> topo = rcm8cube['eta'][-1, :, :]
+        >>> cmsk = dm.mask.ChannelMask(velocity, topo)
 
     And visualize the mask:
-        >>> cmsk.show_mask()
+        >>> cmsk.show()
 
     .. plot:: mask/channelmask.py
 
     """
 
-    def __init__(self, arr, is_mask=False):
+    def __init__(self, velocity, topo, **kwargs):
         """Initialize the ChannelMask.
 
-        Intializing the channel mask requires an array of data, should be
-        two-dimensional.
+        Intializing the channel mask requires a flow velocity field and an
+        array of the delta topography.
+
+        Parameters
+        ----------
+        velocity : ndarray
+            The velocity array to be used for mask creation.
+
+        topo : ndarray
+            The model topography to be used for mask creation.
+
+        Other Parameters
+        ----------------
+        velocity_threshold : float, optional
+            Threshold velocity above which flow is considered 'channelized'.
+            The default value is 0.3 m/s based on DeltaRCM default parameters
+            for sediment transport.
+
+        landmask : :obj:`LandMask`, optional
+            A :obj:`LandMask` object with a defined binary land mask.
+            If given, it will be used to help define the channel mask.
+
+        wetmask : :obj:`WetMask`, optional
+            A :obj:`WetMask` object with a defined binary wet mask.
+            If given, the landmask attribute it contains will be used to
+            determine the channel mask.
+
+        topo_threshold : float, optional
+            Threshold depth to use for the OAM. Default is -0.5.
+
+        angle_threshold : int, optional
+            Threshold opening angle used in the OAM. Default is 75 degrees.
+
+        numviews : int, optional
+            Defines the number of times to 'look' for the OAM. Default is 3.
+
+        is_mask : bool, optional
+            Whether the data in :obj:`arr` is already a binary mask. For
+            example, this should be set to True, if you have already binarized
+            the data yourself, using custom routines, and want to just store
+            the data in the LandMask object.
+
+        """
+        super().__init__(mask_type='channel', data=topo)
+        self.velocity = velocity
+
+        # assign **kwargs to self
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+        # set default values for **kwargs if they do not exist
+        self.velocity_threshold = getattr(self, 'velocity_threshold', 0.3)
+        self.topo_threshold = getattr(self, 'topo_threshold', -0.5)
+        self.angle_threshold = getattr(self, 'angle_threshold', 75)
+        self.numviews = getattr(self, 'numviews', 3)
+        self.is_mask = getattr(self, 'is_mask', False)
+
+        if not self.is_mask:
+            self.compute_channelmask()
+        elif self.is_mask is True:
```

Comment:
I shoudl have said explicitly: these should also be changed, as above, to `elif self.is_mask:`

Reply:
Looking into this further and I think that switching to `elif self.is_mask:` means that any value for `is_mask` other than False is caught in this `elif` statement. I am also thinking that I might switch back to `False` instead of `if not` as I think it is more obvious what the loop is checking when `False` and `True` are explicitly stated (unless you have a strong objection to it).

Reply:
yeah okay, you're probably right: https://docs.quantifiedcode.com/python-anti-patterns/readability/comparison_to_true.html
Seems like `if cond is True:` is fine if you want to be sure it's boolean.

Sorry for the trouble.